OAuthLab/app/src/main/java/com/colinbradley/oauthlab/MainActivity.java

```diff
@@ -0,0 +1,166 @@
+
+
+
+
+        ConnectivityManager connectivityManager = (ConnectivityManager)getSystemService(CONNECTIVITY_SERVICE);
+        final NetworkInfo info = connectivityManager.getActiveNetworkInfo();
+
+        if (info != null && info.isConnected()){
+            String consumerCredentials = CONSUMER_KEY + ":" + CONSUMER_SECRET;
+            mEncodedCrudentials = new String(Base64.encodeBase64(consumerCredentials.getBytes()));
+            getBearerToken();
+        }else {
+            Toast.makeText(this, "NO CONNECTION", Toast.LENGTH_SHORT).show();
+        }
+
+
+        mTweetList = new ArrayList<>();
+        mTweetList.add(new Tweet("placeholder tweet", "time/date"));
+        mRecyclerView = (RecyclerView)findViewById(R.id.recyclerview);
+        mRecyclerView.setLayoutManager(new LinearLayoutManager(MainActivity.this, LinearLayoutManager.VERTICAL,false));
+        mAdapter = new Adapter(mTweetList);
+        mRecyclerView.setAdapter(mAdapter);
+        mEditText = (EditText)findViewById(R.id.edittext);
+        mSearchButton = (Button)findViewById(R.id.searchbutton);
+
+        mSearchButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (info != null && info.isConnected()){
+                    startSearch(mEditText.getText().toString());
+                }
+            }
+        });
+    }
+
+    private void getBearerToken(){
+        OkHttpClient client = new OkHttpClient();
+        Headers headers = new Headers.Builder()
+                .add("Authorization", "Basic " + mEncodedCrudentials)
+                .add("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8")
+                .build();
+        RequestBody body = new FormBody.Builder()
+                .add("grant_type", "client_credentials")
+                .build();
+        final Request request = new Request.Builder()
+                .url("https://api.twitter.com/oauth2/token")
+                .headers(headers)
+                .post(body)
+                .build();
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                try {
+                    JSONObject responseObject = new JSONObject(response.body().string());
+                    mBearerToken = responseObject.getString("access_token");
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
+    private void startSearch(String query){
+        Toast.makeText(this, "Starting Search", Toast.LENGTH_SHORT).show();
+        String url;
+        url = "https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name=" + query;
```

Comment:
Might be helpful to put this url in a constant at the top of the class so it's easier to find